scripts/check_ci_test_suites_to_run.py

```diff
@@ -0,0 +1,689 @@
+    """
+    test_suites_affected: List[GenericTestSuiteDict] = []
+    for test_suite in test_suites:
+        if test_suite['name'] not in test_suites_to_module_mapping:
+            test_suites_affected.append(test_suite)
+    for (
+        test_suite_name,
+        test_suite_modules
+    ) in test_suites_to_module_mapping.items():
+        test_suite_by_name = get_test_suite_by_name_from_list(
+            test_suites,
+            test_suite_name
+        )
+        if test_suite_by_name is not None and (
+            root_file in test_suite_modules or
+            root_file == test_suite_by_name['module']
+        ):
+            test_suites_affected.append(test_suite_by_name)
+
+    return extend_test_suites_without_duplicates(
+        [],
+        test_suites_affected
+    )
+
+
+def get_affected_lighthouse_pages(
+    modified_root_files: List[str]
+) -> List[LighthousePageDict]:
+    """Gets the affected Lighthouse pages by a list of modified root files.
+
+    Args:
+        modified_root_files: list(str). The list of modified root files.
+
+    Returns:
+        list(dict). The affected Lighthouse pages.
+    """
+    lighthouse_pages = get_lighthouse_pages_from_config()
+    affected_lighthouse_pages: List[LighthousePageDict] = []
+    for modified_root_file in modified_root_files:
+        for lighthouse_page in lighthouse_pages:
+            if modified_root_file == lighthouse_page['page_module']:
+                affected_lighthouse_pages.append(lighthouse_page)
+
+    return affected_lighthouse_pages
+
+
+def get_ci_test_suites_to_run(
+    modified_files: List[str],
+    root_files_mapping: dict[str, List[str]]
+) -> CITestSuitesToRunDict | None:
+    """Gets the test suites to run in the CI.
+
+    Args:
+        modified_files: list(str). The list of modified files.
+        root_files_mapping: dict. The root files mapping.
+
+    Returns:
+        dict | None. The test suites to run in the CI or None if all test
+        suites should be run.
+    """
+    modified_root_files = []
+    for file in modified_files:
+        if file not in root_files_mapping:
+            return None
+        for root_file in root_files_mapping[file]:
+            if root_file not in modified_root_files:
+                modified_root_files.append(root_file)
+
+    root_files_config = get_root_files_config()
+    for modified_root_file in modified_root_files:
+        if modified_root_file in root_files_config['run_all_tests_root_files']:
+            return None
+
+    all_test_suites_by_type = get_all_test_suites_by_type()
+
+    acceptance_test_suites_to_module_mapping = (
+        get_test_suites_to_module_mapping_from_directory(
+            os.path.join(
+                TEST_MODULES_MAPPING_DIRECTORY,
+                'acceptance'
+            ),
+            all_test_suites_by_type['acceptance']
+        )
+    )
+
+    acceptance_test_suites: List[GenericTestSuiteDict] = []
```

Comment:
Again use `set` and then you can extend this without having to deal with duplicates.

Reply:
Ditto above.

Reply:
I think I did this similar to not using sets above is because dictionaries are unhashable, and sets are like dictionaries, so you can't have a key being a dictionary, though the case above with strings is fine.